ucp/_libs/core.pyx

```diff
@@ -398,10 +406,16 @@ cdef class ApplicationContext:
 
     def __dealloc__(self):
         if self.initiated:
-            for task in self.progress_tasks:
+            # Aborting all objects working in this context
+            for child in (self.children if self.children else []):
+                child = child()
+                if child is not None:
+                    child.abort()
+            for task in (self.progress_tasks if self.progress_tasks else []):
```

Comment:
Same simplification here 🙂

```suggestion
            for task in self.progress_tasks:
```

Reply:
I added this because sometimes `self.progress_tasks` will be None. We are not setting it to None anywhere so my guess is that it is part of the Python-destructor-order-issue.

Reply:
Thanks for clarifying. Would it make sense to set it to `[]` in those cases instead? Not critical if that seems like too much for this PR.

Reply:
I am not sure that it is possible?

Reply:
Could we add this in an `__init__` or `__cinit__` perhaps?

Reply:
I am not sure I follow. Do you want to wrap `self.progress_tasks` and `self.children` in a class?

Reply:
Am suggesting that `_Listener` have a `__cinit__` where these are initialized with `[]`. Though this is not critical. We can follow-up in another PR if needed. 🙂

Reply:
I see, yeah let's wait with that to another PR.